### DISCUSSION TITLE
Terraform Registry Support

### DISCUSSION THREAD

**Discussion by mdsakalu** [community/community#55948] [Packages]
### Select Topic Area

Product Feedback

### Body

Would be nice if Packages could be used as a [Terraform Registry](https://developer.hashicorp.com/terraform/internals/module-registry-protocol) to publish Terraform modules and providers.

Currently this is the only way to use [semantic versioning constraints](https://developer.hashicorp.com/terraform/language/expressions/version-constraints) when declaring module dependencies.

From [module syntax](https://developer.hashicorp.com/terraform/language/modules/syntax#version):
> Version constraints are supported only for modules installed from a module registry

Packages would be very useful for managing modules, especially for use cases where you don't want them published to the public Terraform Registry.

**Comment by JoeyRichter** [comment]
This is something that GitLab has supported for a while with their [Terraform Module Registry](https://docs.gitlab.com/ee/user/packages/terraform_module_registry/#:~:text=Terraform%20Module%20Registry.%20With%20the#:~:text=Terraform%20Module%20Registry.%20With%20the), and I would love to see GitHub add first-class support as well.
